src/App.js

```diff
@@ -1,5 +1,60 @@
+import {
+  stringifyToJSON,
+  readLineAsync,
+  validatePositiveNumberArray,
+  convertCharacterClassRegex,
+  sumArray,
+  printResult,
+  convertNumberArray,
+} from './utils.js';
+
 class App {
-  async run() {}
+  static DEFAULT_SEPARATOR = ',:';
+  static CUSTOM_SEPARATOR_REGEXP = /\/\/(.+)\\n/;
+
+  async run() {
+    const userInput = await readLineAsync();
+    const processedInput = this.processInput(userInput);
+    const sum = this.calculateSum(processedInput);
+
+    printResult(sum);
+  }
+
+  processInput(input) {
+    if (input === '') return []; // 빈 문자열은 예외로 0을 리턴한다.
+
+    const stringified = stringifyToJSON(input);
+    const separator = this.getSeparator(stringified);
+    const content = this.extractContent(stringified);
+
+    return this.splitContent(content, separator);
+  }
+
+  getSeparator(str) {
+    const customSeparator = str.match(App.CUSTOM_SEPARATOR_REGEXP);
+
+    return customSeparator ? customSeparator[0] : App.DEFAULT_SEPARATOR;
+  }
+
+  extractContent(str) {
+    return str
+      .replace(App.CUSTOM_SEPARATOR_REGEXP, '')
+      .replace(/^"|"$/g, '')
+      .replace('\\\\', '\\');
+  }
+
+  splitContent(content, separator) {
+    const separatorRegExp = convertCharacterClassRegex(separator);
+
+    return content.split(separatorRegExp);
+  }
```

Comment:
```suggestion
  splitContent(content, separator) {
    return content.split(getCharacterClassRegex(separator));
  }
```

convertCharacterClassRegex를 getCharacterClassRegex로 변경한다면 한 줄로 작성해도 가독성이 좋을 거 같아요

Reply:
`get-` prefix는 `인자를 전달하지 않고 결과값을 리턴할 때` 사용하는 것으로 저만의 컨벤션을 지정했어서 `convert-`를 사용했습니다.

`get-`도 좋은 방식이라 생각합니다 !